app/scripts/directives/addSecretToApplication.js

```diff
@@ -0,0 +1,171 @@
+          .concat(replicationControllers)
+          .concat(replicaSets)
+          .concat(statefulSets);
+        ctrl.applications = _.sortBy(apiObjects, ['metadata.name', 'kind']);
+        ctrl.updating = false;
+      }
+    };
+
+    var getApplications = function() {
+      var hasDeploymentFilter = $filter('hasDeployment');
+      var hasDeploymentConfigFilter = $filter('hasDeploymentConfig');
+
+      ctrl.updating = true;
+      var context = {
+        namespace: ctrl.project.metadata.name
+      };
+      // Load all the "application" types
+      DataService.list('deploymentconfigs', context).then(function(deploymentConfigData) {
+        deploymentConfigs = _.toArray(deploymentConfigData.by('metadata.name'));
+        sortApplications();
+      });
+      DataService.list('replicationcontrollers', context).then(function(replicationControllerData) {
+        replicationControllers = _.reject(replicationControllerData.by('metadata.name'), hasDeploymentConfigFilter);
+        sortApplications();
+      });
+      DataService.list({
+        group: 'apps',
+        resource: 'deployments'
+      }, context).then(function(deploymentData) {
+        deployments = _.toArray(deploymentData.by('metadata.name'));
+        sortApplications();
+      });
+      DataService.list({
+        group: 'extensions',
+        resource: 'replicasets'
+      }, context).then(function(replicaSetData) {
+        replicaSets = _.reject(replicaSetData.by('metadata.name'), hasDeploymentFilter);
+        sortApplications();
+      });
+      DataService.list({
+        group: 'apps',
+        resource: 'statefulsets'
+      }, context).then(function(statefulSetData) {
+        statefulSets = _.toArray(statefulSetData.by('metadata.name'));
+        sortApplications();
+      });
+    };
+
+    ctrl.$onInit = function() {
+      ctrl.addType = 'env';
+      ctrl.disableInputs = false;
+      getApplications();
+    };
+
+    ctrl.$postLink = function() {
+      $scope.$watch(function() {
+        return ctrl.application;
+      }, function() {
+        // Look at the existing mount paths so that we can warn if the new value is not unique.
+        var podTemplate = _.get(ctrl.application, 'spec.template');
+        ctrl.existingMountPaths = StorageService.getMountPaths(podTemplate);
+      });
+    };
+
+    ctrl.addToApplication = function() {
+      var podTemplate = _.get(ctrl.application, 'spec.template');
+
+      ctrl.disableInputs = true;
+
+      if (ctrl.addType === 'env') {
+        var newEnvFrom = {
+          secretRef: ctrl.secret.metadata
+        };
+
+        // For each container, add the new volume mount.
+        _.each(podTemplate.spec.containers, function(container) {
+          container.envFrom = container.envFrom || [];
+          container.envFrom.push(newEnvFrom);
+        });
+      } else {
+        var generateName = $filter('generateName');
+        var name = generateName('volume-');
```

Comment:
It might be better to use the secret name as the generateName prefix so it's more obvious what the volume is for.